fir_notifications/methods/jabber.py

```diff
@@ -0,0 +1,88 @@
+import markdown2
+from django.conf import settings
+
+import xmpppy as xmpp
+from django import forms
+
+from fir_notifications.methods import NotificationMethod
+from fir_notifications.methods.utils import request
+from fir_plugins.links import registry as link_registry
+from django.utils.translation import ugettext_lazy as _
+
+
+class Client(xmpp.Client):
+    def __init__(self, *args, **kwargs):
+        kwargs['debug'] = []
+        xmpp.Client.__init__(self, *args, **kwargs)
+
+    def DisconnectHandler(self):
+        pass
+
+
+class XmppMethod(NotificationMethod):
+    use_subject = True
+    use_short_description = True
+    name = 'xmpp'
+    verbose_name = 'XMPP'
+    options = {
+        'jid': forms.CharField(max_length=100, label=_('Jabber ID'))
+    }
+
+    def __init__(self):
+        super(NotificationMethod, self).__init__()
+        self.messages = []
+        self.jid = getattr(settings, 'NOTIFICATIONS_XMPP_JID', None)
+        self.password = getattr(settings, 'NOTIFICATIONS_XMPP_PASSWORD', None)
+        if self.jid is None or self.password is None:
+            self.server_configured = False
+            return
+        self.server = getattr(settings, 'NOTIFICATIONS_XMPP_SERVER', None)
+        self.port = getattr(settings, 'NOTIFICATIONS_XMPP_SERVER_PORT', 5222)
+        self.connection_tuple = None
+        self.use_srv = True
+        self.jid = xmpp.JID(self.jid)
+        if self.server is not None:
+            self.connection_tuple = (self.server, self.port)
+            self.use_srv = False
+        self.client = Client(self.jid.getDomain())
+        if not self.client.connect(server=self.connection_tuple, use_srv=self.use_srv):
```

Comment:
Not sure about this.

What if the server is offline when FIR is started ? It doesn't make sense to me to restart FIR simply to take this fact into account.